internal/transport/handler_server.go

```diff
@@ -141,7 +141,10 @@ type serverHandlerTransport struct {
 	stats []stats.Handler
 }
 
-func (ht *serverHandlerTransport) Close() {
+func (ht *serverHandlerTransport) Close(err error) {
+	if logger.V(logLevel) {
+		logger.Infof("closing serverHandlerTransport due to err: %v", err)
```

Comment:
Same here. A concise error message could look something like
`logger.Infof("Closing server handler transport: %v", err)`

Also, note that log messages should start with a capitalized string unlike error messages. Applies here and everywhere else in this PR (unless ofcourse where it starts with the package name).